json-path/src/main/java/com/jayway/jsonpath/internal/function/MapJsonPathFunctionFactory.java

```diff
@@ -0,0 +1,30 @@
+package com.jayway.jsonpath.internal.function;
+
+import com.jayway.jsonpath.InvalidPathException;
+
+import java.util.Map;
+
+public class MapJsonPathFunctionFactory implements JsonPathFunctionFactory {
+
+    private final Map<String, Class<? extends PathFunction>> pathFunctionMap;
+
+    public MapJsonPathFunctionFactory(
+            final Map<String, Class<? extends PathFunction>> pathFunctionMap) {
+
+        this.pathFunctionMap = pathFunctionMap;
```

Comment:
I suggest you clone the list here so that modifications to the original map don't affect the factory.